src/main/scala/beam/agentsim/agents/ridehail/DefaultRideHailDepotParkingManager.scala

```diff
@@ -0,0 +1,729 @@
+import beam.agentsim.infrastructure.taz.{TAZ, TAZTreeMap}
+import beam.agentsim.scheduler.BeamAgentScheduler.ScheduleTrigger
+import beam.router.BeamRouter.Location
+import beam.router.Modes.BeamMode.CAR
+import beam.router.skim.Skims
+import beam.sim.{BeamServices, Geofence}
+import beam.utils.logging.LogActorState
+import com.typesafe.scalalogging.LazyLogging
+import com.vividsolutions.jts.geom.Envelope
+import org.matsim.api.core.v01.network.Link
+import org.matsim.api.core.v01.{Coord, Id}
+import org.matsim.core.controler.OutputDirectoryHierarchy
+import org.matsim.core.utils.collections.QuadTree
+
+import scala.collection.mutable
+import scala.collection.mutable.ListBuffer
+import scala.util.{Failure, Random, Success, Try}
+import scala.collection.JavaConverters._
+
+/**
+  * Manages the parking/charging depots for the RideHailManager. Depots can contain heterogeneous [[ChargingPlugTypes]]
+  * and any queues for those charger types are tracked separately.
+  *
+  * A Depot is a collection of ParkingZones... each zone represents the combination of a location (i.e. a TAZ) and a ChargingPointType
+  * along with the queue for that charging point type. Queues are managed at the level of a ParkingZone.
+  *
+  * Contains an MNL depot / chargingPlugType selection algorithm that takes into account travel time to the depot (using Skims.od_skimmer),
+  * time spent in queue waiting for a charger, time spent charging the vehicle, and vehicle range. This is a probablistic selection algorithm, so if two
+  * alternatives yield equivalent utility during evaluation, then half of vehicles will be dispatched to each alternative. To decrease
+  * the degree of stochasticity, increase the magnitude of the three MNL params (below) which will decrease the overall elasticity of the
+  * MNL function.
+  *
+  * Key parameters to control behavior of this class:
+  *
+  * beam.agentsim.agents.rideHail.charging.vehicleChargingManager.defaultVehicleChargingManager.mulitnomialLogit.params.drivingTimeMultiplier = "double | -0.01666667" // one minute of driving is one util
+  * beam.agentsim.agents.rideHail.charging.vehicleChargingManager.defaultVehicleChargingManager.mulitnomialLogit.params.queueingTimeMultiplier = "double | -0.01666667" // one minute of queueing is one util
+  * beam.agentsim.agents.rideHail.charging.vehicleChargingManager.defaultVehicleChargingManager.mulitnomialLogit.params.chargingTimeMultiplier = "double | -0.01666667" // one minute of charging is one util
+  * beam.agentsim.agents.rideHail.charging.vehicleChargingManager.defaultVehicleChargingManager.mulitnomialLogit.params.insufficientRangeMultiplier = "double | -60.0" // 60 minute penalty if out of range
+  *
+  * @param parkingFilePath
+  * @param valueOfTime
+  * @param tazTreeMap
+  * @param random
+  * @param boundingBox
+  * @param distFunction
+  * @param parkingStallCountScalingFactor
+  * @param beamServices
+  */
+class DefaultRideHailDepotParkingManager[GEO: GeoLevel](
```

Comment:
@dimaopen please, take a look to this class. It is a replacement of `RideHailDepotParkingManager`